ESCPOS_NET/Extensions/ImageSharpExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+
+namespace SixLabors.ImageSharp
+{
+    public static class ImageSharpExtensions
+    {
+        public static byte[] ToSingleBitPixelByteArray(this Image<Rgba32> image, bool rasterFormat = true, int? maxWidth = null, int? maxHeight = null, float threshold = 0.5F)
+        {
+            image.Mutate(img =>
+            {
+                if (maxWidth.HasValue || maxHeight.HasValue)
+                {
+                    img.Resize(new ResizeOptions
+                    {
+                        Mode = ResizeMode.Max,
+                        Size = new SixLabors.Primitives.Size(maxWidth ?? int.MaxValue, maxHeight ?? int.MaxValue),
+                    });
+                }
+
+                img.BinaryDiffuse(new Processing.Processors.Dithering.StuckiDiffuser(), threshold);
+
+                if (!rasterFormat)
+                {
+                    img.Rotate(RotateMode.Rotate90);
+                    img.Flip(FlipMode.Horizontal);
+                }
+            });
+
+            var result = image.ToSingleBitPixelByteArray();
+            return result;
+        }
+
+        private static byte[] ToSingleBitPixelByteArray(this Image<Rgba32> image)
+        {
+            var bytesPerRow = (image.Width + 7 & -8) / 8;
+
+            var result = new byte[bytesPerRow * image.Height];
+            for (int y = 0; y < image.Height; y++)
+            {
+                var row = image.GetPixelRowSpan(y);
+                var rowStartPosition = y * bytesPerRow;
+
+                for (int x = 0; x < row.Length; x++)
+                {
+                    if (row[x].IsBlack() == false)
```

Comment:
my preference is to not use if statements w/o braces because it can cause maintenance issues if lines are added/removed/reordered, so please define the block here.  Also, if comparing to false, prefer to use a ! instead.